python/tests/test_vars_funs.py

```diff
@@ -122,3 +123,254 @@ def test_vars_rename_invalid_output_type(self):
+        column"""
+        return [
+            {
+                # Structure of the input column. We parameterize input data in
+                # the `input_data` fixture with one parameter per element of
+                # this dict
+                "input": {
+                    "athena": {"name": "pin", "value": ["12345"] * 4},
+                    "iasworld": {"name": "pin", "value": ["12345"] * 4},
+                },
+                # Structure of the output column. We select the proper column
+                # based on key for different types of tests
+                "expected": {
+                    "short": {"name": "pin", "value": ["12345"] * 4},
+                    "long": {"name": "pin", "value": ["12345"] * 4},
+                    "code": {"name": "pin", "value": ["12345"] * 4},
+                    "factor": {"name": "pin", "value": ["12345"] * 4},
+                    # If `value` is True, expect the column value to be
+                    # recoded to the "long" format; otherwise, expect the
+                    # column value to stay the same as the input value
+                    "col": {"name": "pin", "value": False},
+                },
+            },
+            {
+                "input": {
+                    "athena": {
+                        "name": "char_ext_wall",
+                        "value": ["1", "2", "0", None],
+                    },
+                    "iasworld": {
+                        "name": "extwall",
+                        "value": ["1", "2", "0", None],
+                    },
+                },
+                "expected": {
+                    "short": {
+                        "name": "char_ext_wall",
+                        "value": ["FRAM", "MASR", np.nan, np.nan],
+                    },
+                    "long": {
+                        "name": "char_ext_wall",
+                        "value": ["Frame", "Masonry", np.nan, np.nan],
+                    },
+                    "code": {
+                        "name": "char_ext_wall",
+                        "value": ["1", "2", np.nan, np.nan],
+                    },
+                    "factor": {
+                        "name": "char_ext_wall",
+                        "value": pd.Categorical(
+                            ["1", "2", np.nan, np.nan],
+                            categories=["1", "2", "3", "4"],
+                        ),
+                    },
+                    "col": {"name": "char_ext_wall", "value": True},
+                },
+            },
+            {
+                "input": {
+                    "athena": {
+                        "name": "char_bsmt",
+                        "value": ["1", "3", "4", "5"],
+                    },
+                    "iasworld": {
+                        "name": "bsmt",
+                        "value": ["1", "3", "4", "5"],
+                    },
+                },
```

Comment:
**issue (blocking):** We should add a test for numeric values (and update the function to handle those) or make it clear to the user that `vars_recode` expect all values to be strings.